tools/scylla-nodetool.cc

```diff
@@ -918,7 +918,8 @@ void info_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
     auto gossip_running = client.get("/storage_service/gossiping").GetBool();
     fmt::print("{:<23}: {}\n", "ID", rjson::to_string_view(client.get("/storage_service/hostid/local")));
     fmt::print("{:<23}: {}\n", "Gossip active", gossip_running);
-    fmt::print("{:<23}: {}\n", "Thrift active", client.get("/storage_service/rpc_server").GetBool());
+    // thrift protocol is not supported anymore
+    fmt::print("{:<23}: {}\n", "Thrift active", false);
```

Comment:
In theory, the native nodetool can be used to talk to an older version of Scylla, which still has the Thrift api.

Reply:
yes, but i don't think the state of thrift service is important anymore. as it's disabled by default and has been deprecated. what do you think?

Reply:
I think it is still important to present correct information. Was the REST API endpoint removed altogether? Maybe we should undo that change and return hard-coded `false` from there, to keep nodetool working.

Reply:
nodetool works. And the API was not removed. Botond, please take a look at the issue description. It's more of a UX issue.

Reply:
I did read the issue description, but I'm confused. On the one hand you say clients will hit 404, on the other hand you say that nodetool works.

Reply:
It's just a bad timing I think. When nodetool accesses the API , the handler is not registered yet.

Reply:
> no need to backport, this is a UX issue, which surfaces when server is booting.

I see you mention this in the description, I missed this.
I don't think hardcoding `false` to hide this is a good idea. The same problem will exist for the other services -- CQL, alternator, etc.

Reply:
i concur with you. and neither was i satisfied with this change. actually, i put a similar concern at https://github.com/scylladb/scylladb/issues/19923#issuecomment-2268859081 . 

@denesb how about handling the 404 error and print something like "Server is likely starting, please try again later" ?

Reply:
You can get 404 if you have a bug in nodetool and the URL contains a typo. Or the server-side was accidentally changed and the URL was moved. Testing should catch these but still.

See my response at https://github.com/scylladb/scylladb/issues/19923#issuecomment-2272781212.